File: plugins/agent-sdk-dev/examples/e2b-python/src/agent.py
from typing import Optional
from claude_agent_sdk import Agent


def create_agent(
    api_key: str,
    system_prompt: Optional[str] = None,
    model: str = "claude-sonnet-4-5-20250929"
) -> Agent:
    """
    Create a Claude Agent configured for E2B sandbox deployment.

    Args:
        api_key: Anthropic API key
        system_prompt: Custom system prompt for the agent
        model: Claude model to use

    Returns:
        Configured Agent instance
    """

    if system_prompt is None:
        system_prompt = "You are a helpful AI assistant running in a secure E2B sandbox."

    agent = Agent(
        api_key=api_key,
        model=model,
        system_prompt=system_prompt,

        # Configure permissions for sandbox environment
        permissions={
            "bash": True,
            "read": True,
            "write": True,
            "edit": True,
            "web": False  # Disable external web access by default
        },

        # Optional: Add MCP servers for custom tools
        # mcp_servers={
        #     "filesystem": {
        #         "command": "python",
        #         "args": ["mcp_server_filesystem.py"]
        #     }
        # },

        # Optional: Enable streaming
        # streaming=True,

        # Optional: Add custom context
        # additional_context="Running in E2B sandbox environment"
    )

    return agent


async def run_agent(agent: Agent, task: str) -> str:
    """
    Run the agent with a given task.

    Args:
        agent: Configured Agent instance
        task: Task description for the agent

    Returns:
        Agent's response as string
    """
    try:
        result = await agent.run(user_prompt=task)
        return result.output
    except Exception as error:
        print(f"Agent execution error: {error}")
        raise

File: plugins/agent-sdk-dev/examples/e2b-typescript/src/agent.ts
import { Agent } from '@anthropic-ai/claude-agent-sdk'

export interface AgentConfig {
  apiKey: string
  systemPrompt?: string
  model?: string
}

export function createAgent(config: AgentConfig): Agent {
  const {
    apiKey,
    systemPrompt = 'You are a helpful AI assistant running in a secure E2B sandbox.',
    model = 'claude-sonnet-4-5-20250929'
  } = config

  return new Agent({
    apiKey,
    model,
    systemPrompt,

    // Configure permissions for sandbox environment
    permissions: {
      bash: true,
      read: true,
      write: true,
      edit: true,
      web: false // Disable external web access by default
    },

    // Optional: Add MCP servers for custom tools
    // mcpServers: {
    //   filesystem: {
    //     command: 'node',
    //     args: ['node_modules/@anthropic-ai/mcp-server-filesystem/dist/index.js']
    //   }
    // },

    // Optional: Configure streaming
    // streaming: true,

    // Optional: Add custom context
    // additionalContext: 'Running in E2B sandbox environment'
  })
}

export async function runAgent(agent: Agent, task: string): Promise<string> {
  try {
    const result = await agent.run({
      userPrompt: task
    })

    return result.output
  } catch (error) {
    console.error('Agent execution error:', error)
    throw error
  }
}
